### PR TITLE
Use JAliEn credentials in CCDB API

### PR DESCRIPTION
**Pull request by Atlantic777**
This PR adds JAliEn credentials manager to CCDB API and decorates curl handle with JAliEn credentials. Depends on: https://github.com/alisw/alidist/pull/2932

cc @costing @iraklic @yuw726

### CODE REVIEW

**Review by costing — DISMISSED** [review]

**Review by jgrosseo — DISMISSED** [review]

File: CCDB/include/CCDB/CcdbApi.h
Comment (sawenzel) [inline CCDB/include/CCDB/CcdbApi.h:27]: why do we need this?

Comment (Atlantic777) [reply CCDB/include/CCDB/CcdbApi.h:27]: The libjalien isn't a ROOT library and doesn't have a ROOT dictionary. I noticed problems with ROOT interpreter when `CcdbApi.h` was included because it couldn't understand this type.

There is similar code in JAliEn-ROOT / AliEn-ROOT-Legacy, but maybe there is another way around it. In any case, if this header is not explicitly included in a ROOT interpreter session then maybe it works without it.

Comment (jgrosseo) [reply CCDB/include/CCDB/CcdbApi.h:27]: @sawenzel What do you advise?

**Comment by jgrosseo** [comment]
@TimoWilken The CI errors seems unrelated, or am I mistaken?

File: CCDB/include/CCDB/CcdbApi.h
Comment (sawenzel) [reply CCDB/include/CCDB/CcdbApi.h:27]: No hard opinion here. But usually this should not be necessary. One can already use o2::ccdb::CcdbApi in a ROOT session without including the header. And the protection could be put outside as well.
We can also merge as is and remove later.

**Comment by TimoWilken** [comment]
@jgrosseo the libjalienO2 dependency (added in this PR) isn't picked up by QC. ~I'll open a PR in alidist and restart these checks when that's merged.~ While the O2 build runs fine, could you hold off on merging for now so we don't break the QC build?

**Comment by Barthelemy** [comment]
QC should not need to be touched. Dependencies should be transitive in both aliBuild and CMake.

**Comment by Barthelemy** [comment]
I don't have a solution but this is not going to work: 
`   HINTS "${LIBJALIENO2}"`

This variable is unknown in downstream packages. You could use the environment variable `LIBJALIENO2_ROOT`

**Comment by jgrosseo** [comment]
@costing Do you know who could follow this up after @Atlantic777 departure. My understanding is that just the cmake has to be fixed.

**Comment by iraklic** [comment]
@jgrosseo that would probably be me. I read through this thread but I'm not entirely sure what is going on. If you know what needs to be fixed in cmake please let me know and I will have a look.

**Comment by jgrosseo** [comment]
Apart from resolving the merge conflict which is probably simple, follow the comments by @Barthelemy  about the different variable name

**Comment by jgrosseo** [comment]
Thanks for looking into it!

**Comment by yuw726** [comment]
Hi @Barthelemy @jgrosseo @iraklic,
Isn't `${LIBJALIENO2}` passed to cmake in [o2 alidist script](https://github.com/alisw/alidist/blob/master/o2.sh#L178)? Please correct me if I am wrong.

**Comment by Barthelemy** [comment]
@yuw726 the cmake variable `LIBJALIENO2` is set to the value of the environment variable `LIBJALIENO2_ROOT` when configuring `O2`. 

Later in the build, `QualityControl` is built and when it loads the dependencies it tries to use the `FindlibjalienO2.cmake` and fails because, at this stage, `LIBJALIENO2` is not knows. You should try to use the environment variable `LIBJALIENO2_ROOT` in the Find script. Not guaranteed to work but it is worth trying. 

To test do `aliBuild build QualityControl`.

**Comment by jgrosseo** [comment]
@iraklic Any news?

**Comment by iraklic** [comment]
> @iraklic Any news?

Sorry, I am having very busy week. It is on my todo. BTW: do I understand it right that you expect a new PR from me with the conflict in CCDB/src/CcdbApi.cxx resolved or there is some other way of modifying this PR (I have never done that)?

**Comment by jgrosseo** [comment]
> 
> 
> > @iraklic Any news?
> 
> Sorry, I am having very busy week. It is on my todo. BTW: do I understand it right that you expect a new PR from me with the conflict in CCDB/src/CcdbApi.cxx resolved or there is some other way of modifying this PR (I have never done that)?

You cannot push to the branch which made this PR, but you can just start from that branch, push that into your own fork, and then open a PR based on that one. This will include all these changes. We will at that point close this PR.

**Comment by iraklic** [comment]
> > > @iraklic Any news?
> > 
> > 
> > Sorry, I am having very busy week. It is on my todo. BTW: do I understand it right that you expect a new PR from me with the conflict in CCDB/src/CcdbApi.cxx resolved or there is some other way of modifying this PR (I have never done that)?
> 
> You cannot push to the branch which made this PR, but you can just start from that branch, push that into your own fork, and then open a PR based on that one. This will include all these changes. We will at that point close this PR.

Thank you. That's what I was going to do, but wanted to confirm.

**Comment by jgrosseo** [comment]
> 
> 
> > > > @iraklic Any news?
> > > 
> > > 
> > > Sorry, I am having very busy week. It is on my todo. BTW: do I understand it right that you expect a new PR from me with the conflict in CCDB/src/CcdbApi.cxx resolved or there is some other way of modifying this PR (I have never done that)?
> > 
> > 
> > You cannot push to the branch which made this PR, but you can just start from that branch, push that into your own fork, and then open a PR based on that one. This will include all these changes. We will at that point close this PR.
> 
> Thank you. That's what I was going to do, but wanted to confirm.

Any news?

**Comment by jgrosseo** [comment]
Fixed conflict

**Comment by TimoWilken** [comment]
Hi, I've pushed a commit to fix the compilation errors in this PR. (It seems that they were caused by not declaring the new `retreive` and `retrieveFromTFile` methods in the header file.)

**Comment by TimoWilken** [comment]
Hi @jgrosseo, @costing, it seems that my commits have automatically dismissed your reviews. Sorry! Could you re-review the PR so that the CI checks can restart? (The O2 build passes locally for me now, but I haven't done any more testing than that.)

**Review by costing — APPROVED** [review]

**Comment by costing** [comment]
Done, thank you @TimoWilken ! I think it's also requiring a review from @Barthelemy

**Comment by Barthelemy** [comment]
@costing could you give me a bit of context ?

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
> 
> 
> @costing could you give me a bit of context ?

It is fine. I approved it.

**Comment by jgrosseo** [comment]
fullCI is unrelated. I'll ignore the space checker on this one (apologies @vkucera , we'll fix this later), and merge when the other two CIs have passed.

**Comment by Barthelemy** [comment]
I tried to compile and got the following error: 
```
2021-05-27@11:25:43:DEBUG:QualityControl:QualityControl:0: -- The following REQUIRED packages have not been found:
2021-05-27@11:25:43:DEBUG:QualityControl:QualityControl:0: 
2021-05-27@11:25:43:DEBUG:QualityControl:QualityControl:0:  * libjalienO2
2021-05-27@11:25:43:DEBUG:QualityControl:QualityControl:0:    For CCDB API
2021-05-27@11:25:43:DEBUG:QualityControl:QualityControl:0: 
```
The command I used on CC7 is 
```
aliBuild build QualityControl --defaults o2 --debug
```

alidist is up to date with `master`

**Comment by jgrosseo** [comment]
@TimoWilken do you know why QC misses this?

**Comment by iraklic** [comment]
I am trying to understand nuts and bolts of what is going on here, but I am having trouble to compile dev at this point. It crushes with this:
`/home/iraklic/jAliEn_Clean/sw/SOURCES/Monitoring/v3.8.6/v3.8.6/src/Transports/Unix.cxx:18:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
`

I checked out `dev` and building with `aliBuild build O2 --defaults o2`

Any idea?

**Comment by Barthelemy** [comment]
some of your stuff is not up to date. Make sure you update O2

**Comment by iraklic** [comment]
What kind of stuff? I pulled O2 using `aliBuild init O2@dev --defaults o2` and even did `git pull` today just in case and tried to rebuild. But have the same issue.

**Comment by iraklic** [comment]
Is there a mattermost/slack channel or something similar when I can ask about my building error? Or should I use alice-project-analysis-task-force mailing list? Or is there a fail-safe tag for O2 I can check out that will surely build? Any help is appreciated.

**Comment by iraklic** [comment]
OK. looks like I am not alone in this: https://alice.its.cern.ch/jira/browse/O2-2327. But the problem is that the workaround yields an entirely new issue:
```
In file included from /home/iraklic/jAliEn_Clean/sw/SOURCES/O2/master/0/Common/Utils/src/ShmManager.cxx:19:
/home/iraklic/jAliEn_Clean/sw/SOURCES/O2/master/0/Common/Utils/include/CommonUtils/ShmManager.h:40:20: error: field 'counter' has incomplete type 'std::atomic<int>'
   40 |   std::atomic<int> counter = 0; // atomic counter .. counter number of attached processes
      |                    ^~~~~~~
```

Any suggestions are appreciated

**Comment by jgrosseo** [comment]
The JIRA ticket you quote is closed and it is said " This should not be the case anymore, provided you move to alibuild 1.8.4 " . 
Which alibuild do you have?

**Comment by jgrosseo** [comment]
You can ask in O2 release integration or O2 analysis challenge on mattermost any additional questions.

**Comment by iraklic** [comment]
@jgrosseo has this been merged through other PR? I am asking because there are mentions of conflicts mentioned here being fixed and so on. The compilation problem that I had I fixed using the workaround suggested in the ticket that I mentioned. Anyways please let me know if this has been merged though some other PR or if you want me to check it on my own and open new PR with this change. Thanks.

**Comment by jgrosseo** [comment]
@iraklic I don't think this has been merged through another PR. It is just that other changes had been made to the same code and the conflicts need to be resolved.

**Comment by jgrosseo** [comment]
I reopened it and do not see a conflicts actually.
However, there are failures in the CI? Could you update this PR with what needs to be done, so that it goes green and we can merge it?

**Comment by pzhristov** [comment]
I will try to fix the CMake part and provide modifications in the alidist recipes since the CI issues are related to the discovery of the package.

**Comment by pzhristov** [comment]
The following PR https://github.com/alisw/alidist/pull/3275 should solve the issues we observe in the CI.

**Pull request merged**
